src/vs/workbench/electron-sandbox/parts/titlebar/menubarControl.ts

```diff
@@ -126,15 +132,17 @@ export class NativeMenubarControl extends MenubarControl {
 					}
 
 					const menuToDispose = this.menuService.createMenu(menuItem.item.submenu, this.contextKeyService);
-					this.populateMenuItems(menuToDispose, submenu, keybindings);
 
-					let menubarSubmenuItem: IMenubarMenuItemSubmenu = {
-						id: menuItem.id,
-						label: title,
-						submenu: submenu
-					};
+					if (this.populateMenuItems(menuToDispose, submenu, keybindings)) {
```

Comment:
why do we need all the extra change here? we already have the info needed to check if the submenu has no entries after running populate... (submenu.items.length)